diff --git a/src/CiviEntityStorage.php b/src/CiviEntityStorage.php
index b1402db..bc3cb3e 100644
--- a/src/CiviEntityStorage.php
+++ b/src/CiviEntityStorage.php
@@ -456,6 +456,7 @@ class CiviEntityStorage extends SqlContentEntityStorage {
           else {
             $datetime_format = $definition->getSetting('datetime_type') === DateTimeItem::DATETIME_TYPE_DATE ? DateTimeItemInterface::DATE_STORAGE_FORMAT : DateTimeItemInterface::DATETIME_STORAGE_FORMAT;
             $default_timezone = $definition->getSetting('datetime_type') === DateTimeItem::DATETIME_TYPE_DATE ? DateTimeItemInterface::STORAGE_TIMEZONE : \Drupal::config('system.date')->get('timezone.default') ?? date_default_timezone_get();
+            $item_values[$delta]['original_value'] = $item[$main_property_name];
             $datetime_value = (new \DateTime($item[$main_property_name], new \DateTimeZone($default_timezone)))->setTimezone(new \DateTimeZone('UTC'))->format($datetime_format);
             $item_values[$delta][$main_property_name] = $datetime_value;
           }
diff --git a/src/Plugin/views/field/CustomEntityField.php b/src/Plugin/views/field/CustomEntityField.php
index d8763a7..59de32d 100644
--- a/src/Plugin/views/field/CustomEntityField.php
+++ b/src/Plugin/views/field/CustomEntityField.php
@@ -219,22 +219,23 @@ class CustomEntityField extends EntityField {
    *   Returns the processed entity.
    */
   protected function createEntity(EntityInterface $entity) {
-    $processed_entity = clone $entity;
+    $processed_entity = $entity;
 
     try {
-      $result = $this->civicrmApi->get('CustomValue', [
-        'sequential' => 1,
-        'return' => [$this->definition['field_name']],
-        'entity_id' => $entity->id(),
-        'entity_table' => $entity->getEntityTypeId(),
-      ]);
-
+      $result = $processed_entity->{$this->definition['field_name']}->getValue();
       if (!empty($result)) {
         $result = reset($result);
-        $result = array_filter($result, function ($key) {
-          return is_int($key);
-        }, ARRAY_FILTER_USE_KEY);
-
+        if (isset($result['original_value'])) {
+          $result = [$result['original_value']];
+        }
+        elseif (isset($result['value'])) {
+          $result = [$result['value']];
+        }
+        else {
+          $result = array_filter($result, function ($key) {
+            return is_int($key);
+          }, ARRAY_FILTER_USE_KEY);
+        }
         if (!empty($result)) {
           if (isset($result[0]) && is_array($result[0])) {
             $result = reset($result);
