diff --git a/templates/CRM/Case/Form/Activity/ChangeCaseStatus.tpl b/templates/CRM/Case/Form/Activity/ChangeCaseStatus.tpl
index 1049608475..7a9e27362f 100644
--- a/templates/CRM/Case/Form/Activity/ChangeCaseStatus.tpl
+++ b/templates/CRM/Case/Form/Activity/ChangeCaseStatus.tpl
@@ -13,7 +13,7 @@
       <td class="label">{$form.case_status_id.label}</td>
       <td>{$form.case_status_id.html}</td>
     </tr>
-    {if sizeof($linkedCases) > 0}
+    {if count($linkedCases) > 0}
       <tr>
         <td rowspan="2">{ts}Update Linked Cases Status?{/ts}</td>
         <td>{$form.updateLinkedCases.html}</td>
