diff --git a/CRM/Contact/Form/Task/SMSTrait.php b/CRM/Contact/Form/Task/SMSTrait.php
index eb6e92e2a7..128b0ae355 100644
--- a/CRM/Contact/Form/Task/SMSTrait.php
+++ b/CRM/Contact/Form/Task/SMSTrait.php
@@ -79,6 +79,7 @@ trait CRM_Contact_Form_Task_SMSTrait {
         ->addWhere('phone_numeric', '>', 0)
         ->addWhere('phone_type_id:name', '=', 'Mobile')
         ->addOrderBy('is_primary')
+        ->addOrderBy('id', 'DESC')
         ->addSelect('id', 'contact_id', 'phone', 'phone_type_id:name', 'phone_numeric', 'contact_id.sort_name', 'phone_type_id', 'contact_id.display_name');
       if ($this->getSubmittedValue('to')) {
         $phoneGet->addWhere('id', 'IN', explode(',', $this->getSubmittedValue('to')));
diff --git a/CRM/Mailing/BAO/Mailing.php b/CRM/Mailing/BAO/Mailing.php
index cf7412e3fe..64f3e7f6dc 100644
--- a/CRM/Mailing/BAO/Mailing.php
+++ b/CRM/Mailing/BAO/Mailing.php
@@ -200,7 +200,7 @@ class CRM_Mailing_BAO_Mailing extends CRM_Mailing_DAO_Mailing implements \Civi\C
         'phone_not_empty' => CRM_Utils_SQL_Select::fragment()->where("$entityTable.phone != ''"),
         'mailing_id' => CRM_Utils_SQL_Select::fragment()->where("mg.mailing_id = #mailingID"),
         'temp_contact_null' => CRM_Utils_SQL_Select::fragment()->where('temp.contact_id IS null'),
-        'order_by' => CRM_Utils_SQL_Select::fragment()->orderBy("$entityTable.is_primary"),
+        'order_by' => CRM_Utils_SQL_Select::fragment()->orderBy("$entityTable.id"),
       ];
     }
     else {
