diff --git a/CRM/Case/Form/Search.php b/CRM/Case/Form/Search.php
index df38d23fcf..6c0ea5f6d0 100644
--- a/CRM/Case/Form/Search.php
+++ b/CRM/Case/Form/Search.php
@@ -67,7 +67,9 @@ class CRM_Case_Form_Search extends CRM_Core_Form_Search {
    */
   public function preProcess() {
     $this->set('searchFormName', 'Search');
-
+    if(empty($_GET['crmSID'])) {
+      $_GET['crmSID'] = 2;
+    }
     //check for civicase access.
     if (!CRM_Case_BAO_Case::accessCiviCase()) {
       CRM_Core_Error::fatal(ts('You are not authorized to access this page.'));
