diff --git a/CRM/Utils/Number.php b/CRM/Utils/Number.php
index 2edf8c3a60..06afe8f4cf 100644
--- a/CRM/Utils/Number.php
+++ b/CRM/Utils/Number.php
@@ -125,7 +125,7 @@ class CRM_Utils_Number {
     $formatter = new \NumberFormatter($locale ?? CRM_Core_I18n::getLocale(), NumberFormatter::DECIMAL);
     $formatter->setSymbol(\NumberFormatter::DECIMAL_SEPARATOR_SYMBOL, CRM_Core_Config::singleton()->monetaryDecimalPoint);
     $formatter->setSymbol(\NumberFormatter::GROUPING_SEPARATOR_SYMBOL, CRM_Core_Config::singleton()->monetaryThousandSeparator);
-    return $formatter->format($amount);
+    return $formatter->format(str_replace(' ', '', $amount));
   }
 
 }
