diff --git a/lib/Drupal/Core/Form/FormErrorHandler.php b/lib/Drupal/Core/Form/FormErrorHandler.php
index a414439f8f..39c14d5568 100644
--- a/lib/Drupal/Core/Form/FormErrorHandler.php
+++ b/lib/Drupal/Core/Form/FormErrorHandler.php
@@ -148,7 +148,7 @@ class FormErrorHandler implements FormErrorHandlerInterface {
 
         // If this direct child belongs to a group populate the grouping element
         // with the children errors.
-        if (!empty($child['#group'])) {
+        if (!empty($child['#group']) && is_string($child['#group']) ) {
           $parents = explode('][', $child['#group']);
           $group_element = NestedArray::getValue($form, $parents);
           if (isset($group_element['#children_errors'])) {
