diff --git a/core/assets/scaffold/files/htaccess b/core/assets/scaffold/files/htaccess
--- a/core/assets/scaffold/files/htaccess
+++ b/core/assets/scaffold/files/htaccess
@@ -142,6 +142,8 @@
   RewriteCond %{REQUEST_URI} !/core/modules/system/tests/https?\.php
   RewriteCond %{REQUEST_URI} !/core/modules/system/tests/modules/legacy_front_controller_test/test_index.php
+  #kcfinder FIX
+  RewriteCond %{REQUEST_URI} !/libraries/civicrm/packages/kcfinder/.*\.php$
   # Deny access to any other PHP files that do not match the rules above.
   # Specifically, disallow autoload.php from being served directly.
   RewriteRule "^(.+/.*|autoload)\.php($|/)" - [F]