diff --git a/.htaccess b/.htaccess
index 4031da475cf..b27eebf0eec 100644
--- a/.htaccess
+++ b/.htaccess
@@ -141,6 +141,8 @@ AddEncoding gzip svgz
   # Copy and adapt this rule to directly execute PHP files in contributed or
   # custom modules or to run another PHP application in the same directory.
   RewriteCond %{REQUEST_URI} !/core/modules/statistics/statistics\.php$
+  #kcfinder FIX
+  RewriteCond %{REQUEST_URI} !/libraries/civicrm/packages/kcfinder/.*\.php$
   # Deny access to any other PHP files that do not match the rules above.
   # Specifically, disallow autoload.php from being served directly.
   RewriteRule "^(.+/.*|autoload)\.php($|/)" - [F]
diff --git a/core/assets/scaffold/files/htaccess b/core/assets/scaffold/files/htaccess
index 4031da475cf..b27eebf0eec 100644
--- a/core/assets/scaffold/files/htaccess
+++ b/core/assets/scaffold/files/htaccess
@@ -141,6 +141,8 @@ AddEncoding gzip svgz
   # Copy and adapt this rule to directly execute PHP files in contributed or
   # custom modules or to run another PHP application in the same directory.
   RewriteCond %{REQUEST_URI} !/core/modules/statistics/statistics\.php$
+  #kcfinder FIX
+  RewriteCond %{REQUEST_URI} !/libraries/civicrm/packages/kcfinder/.*\.php$
   # Deny access to any other PHP files that do not match the rules above.
   # Specifically, disallow autoload.php from being served directly.
   RewriteRule "^(.+/.*|autoload)\.php($|/)" - [F]
